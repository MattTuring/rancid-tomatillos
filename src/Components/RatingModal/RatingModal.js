import React from 'react'


const RatingModal = ({show, addRating, rating, submit}) => {



    return(
      <article className='modal' onClick={(event) => addRating(event)}>
      <h1>SELECT YOUR RATING FOR THIS MOVIE:</h1>
      <div className='ratings'>
        <span id='1'>1</span>
        <span id='2'>2</span>
        <span id='3'>3</span>
        <span id='4'>4</span>
        <span id='5'>5</span>
        <span id='6'>6</span>
        <span id='7'>7</span>
        <span id='8'>8</span>
        <span id='9'>9</span>
        <span id='10'>10</span>
      </div>
      {rating &&<button className='submit-button' onClick={() => {submit(); show()}} >Submit Rating: {rating}</button>}
      <span className='x' onClick={show}>X</span>
      </article>
    )


}

export default RatingModal;
