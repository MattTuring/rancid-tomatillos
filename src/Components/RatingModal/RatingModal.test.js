import React from 'react';
import { shallow } from 'enzyme';
import { RatingModal } from './RatingModal';

describe('RatingModal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RatingModal
      show={true}
      addRating={jest.fn()}
      submit={jest.fn()}
      rating={4}
      id={1}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  })
})
