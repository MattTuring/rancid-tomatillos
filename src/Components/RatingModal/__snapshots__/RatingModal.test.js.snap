// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingModal should match the snapshot 1`] = `
"<article className=\\"modal\\" onClick={[Function: onClick]}>
  <h1>
    SELECT YOUR RATING FOR THIS MOVIE:
  </h1>
  <div className=\\"ratings\\">
    <span id=\\"1\\">
      1
    </span>
    <span id=\\"2\\">
      2
    </span>
    <span id=\\"3\\">
      3
    </span>
    <span id=\\"4\\">
      4
    </span>
    <span id=\\"5\\">
      5
    </span>
    <span id=\\"6\\">
      6
    </span>
    <span id=\\"7\\">
      7
    </span>
    <span id=\\"8\\">
      8
    </span>
    <span id=\\"9\\">
      9
    </span>
    <span id=\\"10\\">
      10
    </span>
  </div>
  <button className=\\"submit-button\\" onClick={[Function: onClick]}>
    Submit Rating: 
    4
     
  </button>
  <span className=\\"x\\" onClick={true}>
    X
  </span>
</article>"
`;
